backend/qemu.pm

```diff
@@ -851,7 +851,13 @@ sub start_qemu {
 
     # Add parameters from QEMU_APPEND var, if any.
     # The first item will have '-' prepended to it.
-    sp("$vars->{QEMU_APPEND}") if $vars->{QEMU_APPEND};
+    if ($vars->{QEMU_APPEND}) {
+        # Split multiple options, if needed
+        my @spl = split(' -', $vars->{QEMU_APPEND});
+        foreach my $i (@spl) {
+            sp(split(' ', $i));
+        }
```

Comment:
would it work to apply the map function here? Something like:

```perl
map { sp(' ', $_) } split(' -', $vars->{QEMU_APPEND}) if $vars->{QEMU_APPEND};
```

Reply:
@okurz No, since there are multiple splits. And IMO, the 1st solution is better for understanding.
For the unit test, could you point me to existing file(s) or example(s), please?

Reply:
You needed to use 2nd split inside the map. But I would just keep the loop for better readability.

For unit tests there's already `t/18-qemu.t`. Not sure whether we already execute this `start_qemu` inside a unit test. If not, it might be easier to move your actual code into a separate function and just test that.

Reply:
I am looking at adding test, but it will take time to finalize.
@okurz As aarch64 needs this patch, could you please merge this one now, and wait for another PR for the test, please?

Reply:
Tests added

Reply:
@ggardet Please remove the tests from the fullstack as it would increase a bit the time for the tests to run. I guess the right place to add unit tests would be https://github.com/os-autoinst/os-autoinst/blob/master/t/18-qemu.t, maybe @richiejp can shed some light here... since the sp() method may not be that easy to test.

Reply:
@foursixnine I tried to add it to 18-qemu.t, but I am not able to check output values. And we need to do_start_vm to check it, anyway.
Moreover, there are 3 tests out of 4 which are very fast since they just output qemu version, list qemu machines and test an invalid option, so does not start a full real test.
Just tell me if I need to remove the tests to get it committed or if it is fine like this.

Reply:
I understand that you actually want to start qemu to make it work but how much longer does it actually take? the full stack test has a problem that it already runs for quite some time without good feedback to the waiting developer what is going on so what I think would be good if we can achieve is that we have the tests you added in a separate test and as fast as possible, e.g. just startup qemu, check that it does not fail on invalid command line parameters and then exit.

Reply:
@okurz , I can move it to another file, and maybe remove the real run to be faster if really needed. 
Any suggestion for the new file name? XX-qemu-options.t? 
I would like to get it in ASAP. We can tune things afterwards, if needed.

Reply:
All comments addressed in latest push.